web/source/text/processor.ts

```diff
@@ -70,7 +70,7 @@ namespace com.keyman.text {
 
       // All old deadkeys and keyboard-specific cache should immediately be invalidated
       // on a keyboard change.
-      this.keyboardInterface.resetContext();
+      this.resetContext();
```

Comment:
I see here you don't call the target's `resetContext()`. Is this going to cause inconsistencies?

Reply:
We don't do that here because the `Processor` doesn't maintain an active `OutputTarget`.

Referencing the pattern of two `resetContext()` calls:

```
if(outputTarget) {
  outputTarget.resetContext();
}
this.keyman.textProcessor.resetContext();
```

This particular function _is_ the second call.  Note how it's always paired with functions that _do_ have access to the `OutputTarget`.

Why the pattern of two?  `Processor`, as part of web-core, doesn't have access to the DOM; the locations with the `outputTarget.resetContext()` are DOM-aware.  Since presently, KMW relies on its DOM methods to control the current `OutputTarget`, this is necessary to facilitate the split.

Reply:
Okay, I understand from a structural point of view. But I am still not clear how this isn't going to result in a target that has an inconsistent state.

Reply:
In case it helps, web-core is basically following our current embedded model.  The app (or in this case, DOM-aware KMW) tells KMW (in this case, web-core) when context changes.

To be a bit more detailed:

- `Processor` and `KeyboardInterface` like to cache a few things during their operation to boost performance, but whenever the context changes, that needs to be dumped.  (Hence, `resetContext`.)
- web-core will _not_ be responsible for managing the actual context.  It does not track an active element / target.
    - Much like how KMW operates in embedded mode within the apps, web-core assumes a "single input element" exists; it's up to the consumer to map this to whatever actual context should be used.
    - The `outputTarget` is always provided to the `Processor` as part of the `KeyEvent` object on every keystroke.  It is not managed by `Processor`.
    - While in theory, we could pay attention to the instances and note whenever the instance has changed, this isn't sufficient to catch all context-change cases - especially for embedded KMW.
- So, it's DOM-aware KMW's responsibility to manage which `OutputTarget` actually _is_ the context.
    - Note that DOM-aware KMW is also the only part that's calling `Processor.resetContext` - it's a web-core API.
    - Embedded KMW would forward the request from native/app code through its DOM layer.